module/zfs/zvol.c

```diff
@@ -1439,9 +1417,7 @@ zvol_release(struct gendisk *disk, fmode_t mode)
 	zvol_state_t *zv;
 	boolean_t drop_suspend = B_TRUE;
 
-	ASSERT(!MUTEX_HELD(&zvol_state_lock));
-
-	mutex_enter(&zvol_state_lock);
+	rw_enter(&zvol_state_lock, RW_READER);
```

Comment:
Can we assert that zvol_state_lock is not held here ?

Reply:
Same as above.

Reply:
Got it.